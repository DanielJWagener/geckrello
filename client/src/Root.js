// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
