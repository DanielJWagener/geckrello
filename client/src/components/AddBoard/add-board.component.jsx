import React from "react";
import { connect } from "react-redux";
import { addBoard } from "../../actions";

import "./add-board.styles.scss";

class AddBoard extends React.Component {
  state = {
    mode: "prompt",
    boardTitle: "",
    borderColor: "#555",
    spans: 0
  };

  // Inialize element height on grid. Its grid height is based on CSS grid spans, so we fetch that value from the DOM and set the number of spans accordingly
  addBoardRef = React.createRef();

  toggleMode = () => {
    if (this.state.mode === "prompt") {
      this.setState({ mode: "input" });
    } else {
      this.setState({ mode: "prompt" });
    }
  };

  // Submit handlers
  onFormSubmit = event => {
    event.preventDefault();

    // Validate input
    if (!this.state.boardTitle) {
      alert("Please enter a title.");
      return false;
    }

    // Submit input
    this.props.addBoard(this.state.boardTitle);

    // Clear input
    this.setState({ boardTitle: "", inputValid: false, borderColor: "#555" });

    // Change mode back to "prompt"
    this.toggleMode();
  };

  // Controlled input and validation to ensure unique list titles
  handleChange = e => {
    this.setState({
      boardTitle: e.target.value
    });

    e.target.value
      ? this.setState({ borderColor: "#4fa644" })
      : this.setState({ borderColor: "red" });
  };

  render() {
    if (this.state.mode === "prompt") {
      return (
        <div ref={this.addBoardRef} className="addBoard addBoard__prompt">
          <button onClick={this.toggleMode} className="addBoard__prompt-button">
            <span className="addBoard__prompt-plus">+</span>
            <span className="addBoard__prompt-label"> Add new board</span>
          </button>
        </div>
      );
    } else {
      return (
        <div
          className="addBoard addBoard__input"
          ref={this.addBoardRef}
          style={{ gridRowEnd: `span ${this.state.spans}` }}
        >
          <div className="addBoard__form">
            <form onSubmit={this.onFormSubmit} className="form">
              <div className="form__group">
                <input
                  type="text"
                  className="form form__input form__input--validation"
                  value={this.state.listTitle}
                  onChange={this.handleChange}
                  style={{ borderBottomColor: this.state.borderColor }}
                  placeholder="New board title..."
                  autoFocus
                />
              </div>
              <div className="form__group">
                <input
                  type="submit"
                  className="form__submit"
                  value="Add Board"
                />
                <button className="form__cancel" onClick={this.toggleMode}>
                  Cancell
                </button>
              </div>
            </form>
          </div>
        </div>
      );
    }
  }
}

const mapStateToProps = state => {
  return { boards: state.boards };
};

export default connect(mapStateToProps, { addBoard })(AddBoard);
