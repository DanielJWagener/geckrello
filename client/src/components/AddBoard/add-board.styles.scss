@import "../../sass/abstracts/_variables.scss";

.addBoard {
  border-radius: 0.5rem;
  margin-bottom: 1rem;
  height: 100px;
  z-index: 1;

  &__prompt {
    background-color: rgba(0, 0, 0, 0.6);

    &-button {
      font-family: inherit;
      height: 100%;
      width: 100%;
      color: white;
      background-color: transparent;
      border: none;
      cursor: pointer;
      font-size: 2rem;
      display: flex;
      justify-content: center;
      align-items: center;

      &:hover {
        color: #bbb;
      }
    }

    &-plus {
      font-size: 4rem;
      margin-right: 1rem;
      font-weight: 400;
    }
  }

  &__input {
    background-color: $component-background;
    border: 2px solid rgb(177, 177, 177);
  }

  &__form {
    width: 100%;
  }
}

.form {
  padding: 0.5rem;

  &__group:not(:last-child) {
    margin-bottom: 0.5rem;
  }

  &__input {
    display: block;
    width: 100%;
    font-size: 2rem;
    line-height: 3.5rem;
    padding: 0 1rem;
    border-radius: 0.5rem;
    border: none;
    color: #333;

    &--validation {
      border-bottom-style: solid;
      border-bottom-width: 3px;
      border-bottom-color: #555;
    }
  }

  &__submit,
  &__edit {
    display: inline-block;
    padding: 1rem 1.5rem;
    border-radius: 0.5rem;
    border: none;
    font-size: 2rem;
    background-color: $color-secondary;
    color: white;
    cursor: pointer;
    transition: all 0.2s;

    &:focus,
    &:active {
      outline: none;
      transform: translateY(2px);
    }

    &:hover {
      background-color: $color-secondary-dark;
    }
  }

  &__cancel {
    display: inline-block;
    margin-left: 1rem;
    font-size: 1.5rem;
    text-decoration: none;
    color: #333;
    background-color: transparent;
    border: none;
    cursor: pointer;

    &:hover {
      color: red;
    }
  }
}
