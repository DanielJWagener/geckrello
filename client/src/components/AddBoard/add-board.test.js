import { shallow } from "enzyme";
import React from "react";
import AddBoard from "./add-board.component";
import Root from "../../Root";

it("renders AddBoard component", () => {
  expect(
    shallow(
      <Root>
        <AddBoard />
      </Root>
    )
  ).toMatchSnapshot();
});
