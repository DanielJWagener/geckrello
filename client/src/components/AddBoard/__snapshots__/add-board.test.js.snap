// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddBoard component 1`] = `ShallowWrapper {}`;
